canvas/src/atlas.rs

```diff
@@ -67,11 +65,11 @@ impl TextureAtlas {
     store_color!(false)
       .or_else(|| {
         let mut size = self.texture.size();
-        if size.height * 2 <= MAX_SIZE {
+        if size.height * 2 <= Texture::MAX_DIMENSION {
```

Comment:
what about size.height and size.width both needed to expand

Reply:
By design, when a texture is not enough to store one pixel, will try to expend both width and height.  Because expand texture is a expensive op.